### PR TITLE
Converting Normal Types to pulumi.* types accept inputs. 

### PR DESCRIPTION
**Pull request by rshade**
I started to convert the standard types to pulumi.*Input types for a couple of tickets on nginx, and cert-manager

### CODE REVIEW

**Comment by rshade** [comment]
@joeduffy @viveklak I need your advice on what to do with https://github.com/joeduffy/pulumi-go-helmbase/pull/3/files#diff-70f1ecb7be66d191151d77019a75ef0ba0404acdcbaf1aa630099ed46d208771R109, there some logic here I don't know what to do with: https://github.com/joeduffy/pulumi-go-helmbase/blob/master/base.go#L185

**Comment by viveklak** [comment]
> @joeduffy @viveklak I need your advice on what to do with https://github.com/joeduffy/pulumi-go-helmbase/pull/3/files#diff-70f1ecb7be66d191151d77019a75ef0ba0404acdcbaf1aa630099ed46d208771R109, there some logic here I don't know what to do with: https://github.com/joeduffy/pulumi-go-helmbase/blob/master/base.go#L185

Not entirely sure we need the `pulumi` tag values handling for values since they are a bag anyway? A merge of the map should be sufficient? Perhaps I am missing the specifics of your question?

File: base.go [hunk 1]
```diff
@@ -172,18 +172,26 @@ func InitDefaults(args *ReleaseType, chart, repo string, values interface{}) {
...
 	// strongly typed values override the weakly typed map.
+	localMap := make(map[string]interface{})
 	if args.Values == nil {
-		args.Values = make(map[string]interface{})
+		args.Values = pulumi.ToMap(localMap)
+	} else {
+		args.Values.ToMapOutput().ApplyT(func(vals map[string]interface{}) map[string]interface{} {
+			for k, v := range vals {
+				localMap[k] = v
+			}
```
Comment (viveklak) [inline base.go:192]: I think you still also want to merge in the items from `values` (4th parameter to `InitDefaults`).

File: base.go [hunk 2]
```diff
@@ -201,7 +209,7 @@ func InitDefaults(args *ReleaseType, chart, repo string, values interface{}) {
 	}
 
 	// Delete the HelmOptions input value -- it's not helpful and would cause a cycle.
-	delete(args.Values, FieldHelmOptionsInput)
+	delete(localMap, FieldHelmOptionsInput)
```
Comment (viveklak) [inline base.go:212]: Seems it would be simpler to just move this to the apply block above before returning?

File: base.go [hunk 3]
```diff
@@ -172,18 +172,26 @@ func InitDefaults(args *ReleaseType, chart, repo string, values interface{}) {
...
 
 	// Blit the strongly typed values onto the weakly typed values, so that the Helm
 	// Release is constructed properly. In the event a value is present in both, the
 	// strongly typed values override the weakly typed map.
+	localMap := make(map[string]interface{})
 	if args.Values == nil {
-		args.Values = make(map[string]interface{})
+		args.Values = pulumi.ToMap(localMap)
+	} else {
+		args.Values.ToMapOutput().ApplyT(func(vals map[string]interface{}) map[string]interface{} {
```
Comment (viveklak) [inline base.go:189]:
```suggestion
		merged := args.Values.ApplyT(func(vals map[string]interface{}) map[string]interface{} {
```
And then on line 194 `args.Values = merged`?

I haven't checked the types and perhaps some conversions would be required...

**Review by viveklak — CHANGES_REQUESTED** [review]

**Comment by rshade** [comment]
@viveklak and I hacked on it. We decided to leave it a map, and conquer it in a future release.

**Comment by rshade** [comment]
```
Diagnostics:
  pulumi:pulumi:Stack (simple-cert-manager-ts-dev):
    I0513 15:52:53.221111   24105 request.go:665] Waited for 1.055818426s due to client-side throttling, not priority and fairness, request: GET:https://212.2.244.219:6443/apis/storage.k8s.io/v1?timeout=32s

Outputs:
    certManagerStatus: {
        appVersion: "v1.8.0"
        chart     : "cert-manager"
        name      : "cert-manager-helm-20d9a2d3"
        namespace : "sandbox-ns-0e1ef72a"
        revision  : 1
        status    : "deployed"
        version   : "v1.8.0"
    }

Resources:
    + 5 created

Duration: 59s
```

**Review by viveklak — APPROVED** [review]

**Review by joeduffy — APPROVED** [review]

**Pull request merged**